sdk-extensions/resources/src/main/java/io/opentelemetry/sdk/extension/resources/ContainerResource.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.resources;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/** Factory for {@link Resource} retrieving Container ID information. */
+public final class ContainerResource {
+
+  private static final Logger logger = Logger.getLogger(ContainerResource.class.getName());
+  private static final String UNIQUE_HOST_NAME_FILE_NAME = "/proc/self/cgroup";
+  private static final Pattern HEX_EXTRACTOR =
+      Pattern.compile("^([\\w]*?-)?([a-fA-F0-9]+)(\\.[\\w]*?)?$");
+
+  private final String cgroupFilePath;
+
+  // package private for testing purposes
+  ContainerResource(String cgroupFilePath) {
+    this.cgroupFilePath = cgroupFilePath;
+  }
+
+  private static final ContainerResource INSTANCE =
+      new ContainerResource(UNIQUE_HOST_NAME_FILE_NAME);
+
+  /** Returns resource with container information. */
+  public static Resource get() {
+    String containerId = INSTANCE.extractContainerId();
+
+    if (containerId == null) {
+      return Resource.empty();
+    }
+    return Resource.create(Attributes.of(ResourceAttributes.CONTAINER_ID, containerId));
```

Comment:
I'm confused by this bit. I would just have a private static create() method that did this work. Then you don't have to have any logic in here, nor create what is just a copy of the existing INSTANCE on return. 

Take a look at how the EcsResource does this for a good example of the pattern.

Reply:
written to make it in-line with other factories.